### PR TITLE
python3Packages.django-mdeditor: patch out polyfill.io usage, bump KaTeX

### PR DESCRIPTION
**Pull request by tomodachi94**
Part of #323379.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tomodachi94** [comment]
Also requesting a review from @LeSuisse, who has been largely responsible for most of the other fixes in upstreams and Nixpkgs.

**Review by LeSuisse — CHANGES_REQUESTED** [review]
Since the change is in a non merged upstream PR the patch should be vendored in nixpkgs so we can be sure it does not disappear.

Also the upstream PR likely require some changes, the URLs used as a replacement do not seem to work:
```
$ curl https://cdn.jsdelivr.net/npm/katex@0.3.0/dist/katex.min.css
Couldn't find the requested file /dist/katex.min.css in katex.
$ curl https://cdn.jsdelivr.net/npm/katex@0.3.0/dist/katex.min.js
Couldn't find the requested file /dist/katex.min.js in katex.
```

**Comment by tomodachi94** [comment]
I've copied the patch in-tree. I had to bump KaTeX, since the CSS wasn't available in the NPM package (that jsdelivr uses) until a later version (I picked 0.7.1, since that was the latest that retained backwards compatibility with 0.3.0).

File: pkgs/development/python-modules/django-mdeditor/Bump-KaTeX-and-replace-bootcdn-with-jsdelivr.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+index be0005d..8aacb56 100644
+--- a/mdeditor/static/mdeditor/js/editormd.js
++++ b/mdeditor/static/mdeditor/js/editormd.js
+@@ -4179,8 +4179,8 @@
+     // 使用国外的CDN，加载速度有时会很慢，或者自定义URL
+     // You can custom KaTeX load url.
+     editormd.katexURL  = {
+-        css : "//cdn.bootcdn.net/ajax/libs/KaTeX/0.3.0/katex.min",
+-        js  : "//cdn.bootcdn.net/ajax/libs/KaTeX/0.3.0/katex.min"
++        css : "//cdn.bootcdn.net/ajax/libs/KaTeX/0.7.1/katex.min",
```
Comment (LeSuisse) [inline pkgs/development/python-modules/django-mdeditor/Bump-KaTeX-and-replace-bootcdn-with-jsdelivr.patch:23]: I'm guessing you to depend on cdn.jsdelivr.net here instead :)

Comment (tomodachi94) [reply pkgs/development/python-modules/django-mdeditor/Bump-KaTeX-and-replace-bootcdn-with-jsdelivr.patch:23]: Nope. I split the version bump and the URL change into two separate commits. There are two patches in this patch.

Comment (tomodachi94) [reply pkgs/development/python-modules/django-mdeditor/Bump-KaTeX-and-replace-bootcdn-with-jsdelivr.patch:23]: See further down in the file: https://github.com/tomodachi94/nixpkgs/blob/ced695af7883191ea40a96b5304e5794d3ce8110/pkgs/development/python-modules/django-mdeditor/Bump-KaTeX-and-replace-bootcdn-with-jsdelivr.patch#L35

Comment (LeSuisse) [reply pkgs/development/python-modules/django-mdeditor/Bump-KaTeX-and-replace-bootcdn-with-jsdelivr.patch:23]: Yup sorry, I just misread the patch :)

**Review by LeSuisse — APPROVED** [review]

**Pull request merged**